test/jdk/sun/security/pkcs12/P12SecretKey.java

```diff
@@ -66,6 +66,7 @@ private void run(String keystoreType, String algName, int keySize) throws Except
         ks.setEntry(ALIAS, ske, kspp);
 
         File ksFile = File.createTempFile("test", ".test");
+        ksFile.deleteOnExit();
```

Comment:
Is there a reason why this change is done differently than the rest of the changes in this PR? Can we delete the `File` in a try/finally instead of enrolling a shutdown hook to delete it?